tools/svaba.cwl

```diff
@@ -1,53 +1,88 @@
-cwlVersion: v1.0
+cwlVersion: v1.2
 class: CommandLineTool
 id: svaba
+doc: |
+  SvABA is a method for detecting structural variants in sequencing data using
+  genome-wide local assembly. Under the hood, SvABA uses a custom implementation
+  of SGA (String Graph Assembler) by Jared Simpson, and BWA-MEM by Heng Li.
+  Contigs are assembled for every 25kb window (with some small overlap) for every
+  region in the genome. The default is to use only clipped, discordant, unmapped
+  and indel reads, although this can be customized to any set of reads at the
+  command line using VariantBam rules. These contigs are then immediately aligned
+  to the reference with BWA-MEM and parsed to identify variants. Sequencing reads
+  are then realigned to the contigs with BWA-MEM, and variants are scored by
+  their read support.
+
+  SvABA is currently configured to provide indel and rearrangement calls (and
+  anything "in between"). It can jointly call any number of BAM/CRAM/SAM files,
+  and has built-in support for case-control experiments (e.g. tumor/normal, or
+  trios or quads). In case/control mode, any number of cases and controls (but
+  min of 1 case) can be input, and will jointly assemble all sequences together.
+  If both a case and control are present, variants are output separately in
+  "somatic" and "germline" VCFs. If only a single BAM is present (input with the
+  -t flag), a single SV and a single indel VCF will be emitted.
+
+  A BWA-MEM index reference genome must also be supplied with -G.
 requirements:
   - class: InlineJavascriptRequirement
   - class: DockerRequirement
     dockerPull: 'pgc-images.sbgenomics.com/d3b-bixu/svaba:1.1.0'
   - class: ResourceRequirement
-    ramMin: $(inputs.ram)
+    ramMin: $(inputs.ram * 1000)
     coresMin: $(inputs.cores)
 baseCommand: ['svaba','run']
+arguments:
+  - position: 0
+    valueFrom: "--g-zip"
 inputs:
-  tumor_bams: { type: {type: 'array', items: 'File', inputBinding: { prefix: '--tumor-bam' } }, secondaryFiles: [^.bai], inputBinding: { position: 0 }, doc: "Case BAM/CRAM/SAM file (eg tumor). Can input multiple." }
```

Comment:
Hmmm, does `'File[]'` not work here?

Reply:
Because I want each item in the array to have its own input binding. This approach is the community accepted one. I think itemSeparator could work but it has been finicky for me in the past.